### PR TITLE
Code optimisations and clippy warnings fix

### LINKED ISSUE #29: Fix Clippy warnings
There are a bunch of clippy warnings at the current stage of the project.

Do `cargo clippy` to see them. The warnings also come with probable solutions to fix. 

Fix them.

### PR DESCRIPTION
**Pull request by wthrajat**
The PR aims to fix #29.
To keep things simple and commits undo-able, I am making one commit for one fix.
Also, this PR is **work in progress** as there are still many warnings remaining to be fixed. Feel free to point out any mistakes.
Thanks.

### CODE REVIEW

**Comment by wthrajat** [comment]
Some warnings needs to be carefully looked on, as blindly doing what the rust compiler tells us can lead to unwanted problems.
For example, the fix to
```bash
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
    = note: `#[warn(clippy::ptr_arg)]` on by default
```
introduced 5 new problems in the `taker.rs` file. The fix was to replace `this_maker_contract_txes: &Vec<Transaction>` -> `this_maker_contract_txes: &[Transaction]`
The 5 problems are listed in the screenshot below:
2023-11-04-22_57_40-screenshot

The commit has been reverted in this [new commit](https://github.com/utxo-teleport/teleport-transactions/pull/44/commits/6ec18db0e434391c57aedea3d614535194d491a1)

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
tACK 16 ae4a3b5d7da7b665a80b92ee8137abe57f84ab.

Thanks for the work. I feel this chunk of the changes can be merged first and then rest of the clippy warnings can be done in separate PRs.

Just one nit fix request.

Also `cargo fmt` is failing, minor changes. Rest all good to go.

**Comment by rajarshimaitra** [comment]
> Some warnings need to be carefully looked at, as blindly doing what clippy tells us can lead to unwanted problems.

The remaining warnings are of two types.

- Regular clippy warnings. Easy to fix.
- Function/Type definition is too complex. These require deeper thoughts.

So, I suggest creating another PR after this one with the remaining regular fixes.

Create a separate PR to handle all the complex cases.

File: src/maker/mod.rs [hunk 1]
```diff
@@ -189,7 +189,7 @@ pub async fn start_maker_server(maker: Arc<Maker>) -> Result<(), MakerError> {
                 select! {
                     readline_ret = reader.read_line(&mut line) => {
                         match readline_ret {
-                            Ok(n) if n == 0 => {
+                            Ok(0) => {
                                 log::info!("[{}] Connection closed by peer", maker_clone.config.port);
                                 break;
                             }
```
[the remark this answers is not in the document]
Comment (wthrajat) [reply src/maker/mod.rs:195]: @rajarshimaitra
Done [here](https://github.com/utxo-teleport/teleport-transactions/pull/44/commits/3d7b6e6f2c279f289f7b696f2a2bea553170d238#diff-fd2650f1984a9f0789915a8e8c1bac5cc2a16778f3acb0efec2ba30c93e38b64R196)

**Comment by wthrajat** [comment]
@rajarshimaitra With this [commit](https://github.com/utxo-teleport/teleport-transactions/pull/44/commits/3d7b6e6f2c279f289f7b696f2a2bea553170d238), most of the warnings are now fixed.
Only a few warnings like type complexity are remaining, and as suggested, they deserve a new PR under a new issue.

Regarding `cargo fmt` failing, after running `cargo +nightly fmt --check`, I refactored an import as can be seen in this [commit](https://github.com/utxo-teleport/teleport-transactions/pull/44/commits/1343c96333dbf2caf5c0ce45407d26f9fc1112ba).

As of this commit, there are no build errors and no formatting errors.

**Comment by rajarshimaitra** [comment]
fmt lints still missing..

**Pull request merged**